pkg/infoschema/sieve.go

```diff
@@ -75,6 +75,14 @@ func (s *Sieve[K, V]) SetCapacity(capacity uint64) {
 	s.capacity = capacity
 }
 
+func (s *Sieve[K, V]) SetCapacityAndWaitEvict(capacity uint64) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	for s.size > s.capacity {
+		s.evict()
```

Comment:
This may block the access to cache if the difference is large, is there any suggestion for this risk?

Reply:
maybe use a loop and in each round, acuqire the lock, evict some number of items (like in line 102 ~ 104), release the lock.